scipy/spatial/transform/rotation.pyx

```diff
@@ -2140,6 +2140,37 @@ cdef class Rotation:
 
         return self.__class__(np.asarray(self._quat)[indexer], normalize=False)
 
+    def __setitem__(self, indexer, value):
+        """Set rotation(s) at given index(es) from object.
+
+        Parameters
+        ----------
+        indexer : index, slice, or index array
+            Specifies which rotation(s) to replace. A single indexer must be
+            specified, i.e. as if indexing a 1 dimensional array or list.
+
+        value : `Rotation` instance
+            The rotations to set.
+
+        Raises
+        ------
+        TypeError if the instance was created as a single rotation.
+
+        Notes
+        -----
+
+        .. versionadded:: 1.8.0
+        """
+        if self._single:
+            raise TypeError("Single rotation is not subscriptable.")
+
+        if not isinstance(value, Rotation):
+            raise TypeError("value must be a Rotation object")
+
+        quat = np.asarray(self._quat)
```

Comment:
For my understanding, why can't `self._quat` be used directly?

Reply:
`self._quat` has a  `cdef double[:, :]` type (since the cython rewrite) which can't be indexed in the same way as a `np.array`. The `np.asarray` permits normal python indexing such as slicing.

Reply:
OK, thanks, makes sense. Cython types/views are tricky.

I don't have any other comments. Looks good to me.